Python/generated_cases.c.h

```diff
@@ -3766,35 +3763,37 @@
                 value = GETLOCAL(oparg);
                 assert(value != NULL);
                 Py_INCREF(value);
-                STACK_GROW(1);
-                POKE(1, value);
+                _tmp_1 = value;
             }
+            POKE(1, _tmp_1);
             DISPATCH();
         }
 
         TARGET(STORE_FAST__STORE_FAST) {
+            PyObject *_tmp_1 = PEEK(2);
+            PyObject *_tmp_2 = PEEK(1);
```

Comment:
Just a note on the generated code... this looks sort of odd to me. `_tmp_1` is the *second* item on the stack, and is used by the *second* instruction part. `_tmp_2` is the *first* item on the stack and is used by the *first* instruction part.

Not a big deal, but it seems like they should be named in the order they are actually used, like the other superinstructions? My brain took a second to parse and re-parse the generated code a couple of times.

Reply:
I agree this is annoying, but it's hard to get the numbering consistent since there may be variables that aren't popped off the stack but are pushed back onto it. I could number them in reverse order, but there could still be cases where the numbering would be off.

E.g. for stack effect `(a, b -- c)` we'd see something like
```
PyObject *_tmp_3 = PEEK(2);
PyObject *_tmp_2 = PEEK(1);
PyObject *_tmp_1;
{
    PyObject *a = _tmp_3;
    PyObject *b = _tmp_2;
    PyObject *c;
    <c = a + b>
    _tmp_1 = c;
}
STACK_GROW(1);
PEEK(1, _tmp_1);
```
At some point I considered using letters to number the temp variables, but that wouldn't help much.

Once we have a register VM it'll be less of an issue. :-)